src/rust/engine/src/externs/interface.rs

```diff
@@ -1645,8 +1647,8 @@ fn write_digest(
         .materialize_directory(
           destination.clone(),
           lifted_digest,
+          true, // Force everything we write to be mutable
```

Comment:
This should likely eventually be exposed as an option, but this makes sense for now.